File: 1448. Count Good Nodes in Binary Tree/sol.py
class TreeNode:
    def __init__(self, val: int = 0, left: 'Optional[TreeNode]'= None, right : 'Optional[TreeNode]' = None):
        self.val = val
        self.left = left 
        self.right = right 
        
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0
        if not root:
            return res
        stack = [[root, root.val]]
        while len(stack) > 0 :
            node, highest = stack.pop()
            if node:
                if node.val >= highest:
                    res += 1
                stack.append([node.left, max(node.val, highest)])
                stack.append([node.right, max(node.val, highest)])
        return res

File: 1448. Count Good Nodes in Binary Tree/sol.go
package countGoodNodesInBinaryTree

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func goodNodes(root *TreeNode) int {
	if root == nil {
		return 0
	}
	res := 0

	stack := []pair{pair{root, root.Val}}
	for len(stack) > 0 {
		index := len(stack) - 1
		curr := stack[index]
		stack = stack[:index]
		if curr.Node != nil {
			if curr.Node.Val >= curr.Highest {
				res++
			}
			stack = append(stack, pair{curr.Node.Left, max(curr.Node.Val, curr.Highest)}, pair{curr.Node.Right, max(curr.Node.Val, curr.Highest)})
		}
	}
	return res

}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

type pair struct {
	Node    *TreeNode
	Highest int
}
